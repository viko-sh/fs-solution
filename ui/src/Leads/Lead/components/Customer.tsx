import React from 'react';
import moment from 'moment';
import { LeadProps } from '../type';
import { CustomeWrapper, CustomerIcon, CustomerDetails } from '../style';

export const Customer = ({ contactName, createdAt }: Partial<LeadProps>) => {
  return (
    <CustomeWrapper>
      <CustomerIcon>
        {contactName && <span>{contactName[0]}</span>}
      </CustomerIcon>
      <CustomerDetails>
        {contactName}
        <br />
        <span>{moment(createdAt).format('MMMM D @ h:mm a')}</span>
      </CustomerDetails>
    </CustomeWrapper>
  );
};
