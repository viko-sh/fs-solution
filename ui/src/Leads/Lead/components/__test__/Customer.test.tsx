import React from 'react';
import { mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

import { Customer } from '../Customer';

const defaultProps = {
  contactName: 'Darth',
  createdAt: '2019-11-21T18:57:43.000Z'
};

const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = mount(<Customer {...props} />);
  return { props, wrapper };
};

describe('Customer component', () => {
  it('Should render header', done => {
    const { wrapper } = mountComponent();
    expect(wrapper).toHaveLength(1);
    done();
  });
  it('should be match with snapshot', done => {
    // Arrange (no Act required)
    const { wrapper } = mountComponent();
    // Assert
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
