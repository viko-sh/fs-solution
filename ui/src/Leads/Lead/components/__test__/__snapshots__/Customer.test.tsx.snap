// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer component should be match with snapshot 1`] = `ReactWrapper {}`;
